src/profile/helpers/__tests__/updateMutationVariables.test.ts

```diff
@@ -106,112 +108,251 @@ const newAddress = {
-  ]);
-});
+describe('Update variables for basic data (names) are formed correctly', () => {
+  let formValues: Partial<FormValues>;
 
-test('add arrays are empty when using existing data', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    {
-      ...formValues,
-      addresses: [primaryAddress, updatedSecondaryAddress],
-      emails: [primaryEmail, updatedSecondaryEmail],
-      phones: [primaryPhone, updatedSecondaryPhone],
-    },
-    myProfile
-  );
-
-  expect(variables.input.profile.addPhones).toEqual([]);
-  expect(variables.input.profile.addEmails).toEqual([]);
-  expect(variables.input.profile.addAddresses).toEqual([]);
-});
+  beforeEach(() => {
+    formValues = getFormValues(nameFormParts);
+  });
 
-test('update arrays are formed correctly with updated data objects', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    formValues,
-    myProfile
-  );
-  expect(variables.input.profile.updateAddresses).toEqual([
-    getAddressAsComparisonObject(updatedSecondaryAddress),
-  ]);
-  expect(variables.input.profile.updateEmails).toEqual([
-    getEmailAsComparisonObject(updatedSecondaryEmail),
-  ]);
-  expect(variables.input.profile.updatePhones).toEqual([
-    getPhoneAsComparisonObject(updatedSecondaryPhone),
-  ]);
+  test('Variables in update data match formValues even if not changed', () => {
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      formValues,
+      myProfile
+    );
+    expect(variables.input.profile).toEqual(formValues);
+  });
+
+  test('Variables in update data match changes', () => {
+    const newData = {
+      firstName: 'test-firstName',
+      nickname: 'test-nickname',
+      lastName: 'test-lastName',
+    };
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      newData,
+      myProfile
+    );
+    expect(variables.input.profile).toEqual(newData);
+
+    const partialData1 = {
+      firstName: 'test-firstName',
+    };
+    expect(
+      updateMutationVariables(partialData1, myProfile).input.profile
+    ).toEqual(partialData1);
+
+    const partialData2 = {
+      nickname: 'test-nickname',
+      lastName: 'test-lastName',
+    };
+    expect(
+      updateMutationVariables(partialData2, myProfile).input.profile
+    ).toEqual(partialData2);
+  });
 });
 
-test('update arrays are empty when using unchanged or new data', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    {
-      ...formValues,
-      addresses: [primaryAddress, newAddress],
-      emails: [primaryEmail, newEmail],
-      phones: [primaryPhone, newPhone],
-    },
-    myProfile
-  );
-
-  expect(variables.input.profile.updateAddresses).toEqual([]);
-  expect(variables.input.profile.updateEmails).toEqual([]);
-  expect(variables.input.profile.updatePhones).toEqual([]);
+describe('Update variable for language is formed correctly', () => {
```

Comment:
Minor notice about language: It seems in some of the tests the first letter is capitalised and in some tests it's not. Not a big thing though

Reply:
Yes, sometimes it's a typo, usually the case is:

describe("This describes the purpose of the test group ')
it('and this is in lower case becasue it continues the sentence started in describe()")